### PR TITLE
resource_group: don't accumulate tokens when burstlimit less than 0

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #7206

### What is changed and how does it work?
We can find a large number from default resource group.
```
{"r_u":{"settings":{"fill_rate":2147483647,"burst_limit":-1},"state":{"tokens":29860685960413220,"last_update":"2023-12-27T08:19:16.269363735Z","initialized":true}}}

```
So don't accumulate tokens when burstlimit less than 0

```commit-message
```

### Check List

Tests

- Unit test

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by CabinfeverB** [comment]
@nolouch do we need cherry-pick?

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by nolouch** [comment]
> @nolouch do we need cherry-pick?

Yes, I think we need.

File: pkg/mcs/resourcemanager/server/token_buckets.go [hunk 1]
```diff
@@ -291,21 +291,23 @@ func (gtb *GroupTokenBucket) updateTokens(now time.Time, burstLimit int64, clien
...
-	} else if delta := now.Sub(*gtb.LastUpdate); delta > 0 {
-		elapseTokens = float64(gtb.Settings.GetFillRate())*delta.Seconds() + gtb.lastBurstTokens
-		gtb.lastBurstTokens = 0
-		gtb.Tokens += elapseTokens
-		gtb.LastUpdate = &now
+	} else if burst := float64(burstLimit); burst > 0 {
+		if delta := now.Sub(*gtb.LastUpdate); delta > 0 {
+			elapseTokens = float64(gtb.Settings.GetFillRate())*delta.Seconds() + gtb.lastBurstTokens
+			gtb.lastBurstTokens = 0
+			gtb.Tokens += elapseTokens
```
Comment (bufferflies) [inline pkg/mcs/resourcemanager/server/token_buckets.go:298]: Do you need to update the properties of `LastUpdate` also?

**Review by bufferflies — APPROVED** [review]

**Pull request merged**